File: src/main/java/com/fish/entity/pojo/TopicSolutionsComments.java
package com.fish.entity.pojo;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 *  实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(value = "oj_topic_solutions_comments")
public class TopicSolutionsComments implements Serializable {

    @Id(keyType = KeyType.Auto)
    private Long commentId;

    private String userId;

    private Long solutionId;

    /**
     * 父评论的id，作为鉴别是否为二级评论的标志
     */
    private Long parentId;

    private String content;

    private Boolean enabled;
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime createTime;

}

File: src/main/kotlin/com/fish/entity/pojo/TopicSolutionsComments.kt
package com.fish.entity.pojo

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.KeyType
import com.mybatisflex.annotation.Table
import java.io.Serializable
import java.time.LocalDateTime

/**
 * 实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Table(value = "oj_topic_solutions_comments")
open class TopicSolutionsComments(
    commentId: Long?,
    userId: String?,
    solutionId: Long?,
    parentId: Long?,
    content: String?,
    enabled: Boolean?,
    createTime: LocalDateTime?
) : Serializable {
    @Id(keyType = KeyType.Auto)
    val commentId: Long? = null
    val userId: String? = null
    val solutionId: Long? = null
    /**
     * 父评论的id，作为鉴别是否为二级评论的标志
     */
    val parentId: Long? = null
    val content: String? = null
    val enabled: Boolean? = null
    @JsonSerialize(using = LocalDateTimeSerializer::class)
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    val createTime: LocalDateTime? = null
}
